dear_petition/petition/models.py

```diff
@@ -161,16 +161,22 @@ def refresh_record_from_data(self):
         # make changes to a users a offenses (I doubt it) then we need
         # think about revising the methods chosen here.
         offenses = self.data.get("Offense Record", {})
+
+        # Check and remove any existing Offense Records and the existing Offenses
+        # before reading from raw_data
+        existing_offense_records = OffenseRecord.objects.filter(
+            offense=self.offenses.first()
```

Comment:
Since there can be multiple `Offense`s for a `CIPRSRecord`, we should make sure to handle all of them here (something like `offense__ciprs_record=self)`

Reply:
how interesting. I didn't know you could do that. done